app/helpers/application_helper/button/host_aggregate_delete.rb

```diff
@@ -0,0 +1,10 @@
+class ApplicationHelper::Button::HostAggregateDelete < ApplicationHelper::Button::Basic
+  def calculate_properties
+    super
+    self[:title] = @record.unsupported_reason(:delete_aggregate) if disabled?
+  end
+
+  def disabled?
+    !@record.supports?(:delete_aggregate)
+  end
+end
```

Comment:
we have a class for this: `ApplicationHelper::Button::GenericFeatureButton` or the the one with text. 

See ApplicationHelper::Button::GenericFeatureButton

Unless I am mistaken you don't need this file nore the file below.